tests/integration/test_credentials.py

```diff
@@ -0,0 +1,106 @@
+# Copyright 2015 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+import os
+import mock
+
+from botocore.session import Session
+from tests import unittest
+
+
+class TestCredentialPrecedence(unittest.TestCase):
+    def setUp(self):
+        # Clean up any existing environment
+        for name in ['AWS_ACCESS_KEY_ID',
+                     'AWS_SECRET_ACCESS_KEY',
+                     'BOTO_DEFAULT_PROFILE']:
+            if name in os.environ:
+                del os.environ[name]
```

Comment:
This is general test case best practice advice:  Do *not* leave global state modified outside the scope of a test.  It's ok to mess with global state in a test if there's no better way to test what you need to test, but you *have* to put the state back to the way it initially was.

Here, if the vars you're deleting were defined in os.environ before the test, this setUp() is deleting these values and not restoring them after the test executes.  This can lead to all sorts of hard to track down problems where the test order may affect whether or not a test passes depending on if it passes before/after the global state modification.

That being, said, just use `BaseEnvVar` to handle the patching of env vars for you.